lib/compress.js

```diff
@@ -153,10 +153,7 @@ merge(Compressor.prototype, {
     AST_Node.DEFMETHOD("clear_opt_flags", function(){
         this.walk(new TreeWalker(function(node){
             if (node instanceof AST_SymbolRef) {
-                var d = node.definition();
-                if (d && d.init) {
```

Comment:
There are [many](https://github.com/mishoo/UglifyJS2/blob/0610c020b1544820be9898a285ab6c9066490552/lib/compress.js#L1439-L1440) [places](https://github.com/mishoo/UglifyJS2/blob/0610c020b1544820be9898a285ab6c9066490552/lib/compress.js#L1466-L1467) [in](https://github.com/mishoo/UglifyJS2/blob/0610c020b1544820be9898a285ab6c9066490552/lib/compress.js#L1498) [the](https://github.com/mishoo/UglifyJS2/blob/0610c020b1544820be9898a285ab6c9066490552/lib/compress.js#L1498) codebase where `AST_Symbol.definition()` is assumed to return defined, non-null value.